lib/improver/tests/orographic_enhancement/test_OrographicEnhancement.py

```diff
@@ -0,0 +1,839 @@
+
+
+class Test__init__(IrisTest):
+    """Test the __init__ method"""
+
+    def test_basic(self):
+        """Test initialisation with no arguments"""
+        plugin = OrographicEnhancement()
+        self.assertAlmostEqual(plugin.orog_thresh_m, 20.)
+        self.assertAlmostEqual(plugin.rh_thresh_ratio, 0.8)
+        self.assertAlmostEqual(plugin.vgradz_thresh_ms, 0.0005)
+        self.assertAlmostEqual(plugin.upstream_range_of_influence_km, 15.)
+        self.assertAlmostEqual(plugin.efficiency_factor, 0.23265)
+        self.assertAlmostEqual(plugin.cloud_lifetime_s, 102.)
+
+
+class Test__repr__(IrisTest):
+    """Test the __repr__ method"""
+
+    def test_basic(self):
+        """Test string representation of plugin"""
+        expected = ('OrographicEnhancement() instance with orography '
+                    'threshold 20.0 m, relative humidity threshold 0.8, '
+                    'v.gradz threshold 0.0005 m/s, maximum upstream influence '
+                    '15.0 km, upstream efficiency factor 0.23265, cloud '
+                    'lifetime 102.0 s')
+        plugin = OrographicEnhancement()
+        self.assertEqual(str(plugin), expected)
+
+
+class Test__orography_gradients(IrisTest):
+    """Test the _orography_gradients method"""
+
+    def setUp(self):
+        """Set up an input cube"""
+        self.plugin = OrographicEnhancement()
+        data = np.array([[200., 450., 850.],
+                         [320., 500., 1000.],
+                         [230., 600., 900.]])
+        x_coord = DimCoord(np.arange(3), 'projection_x_coordinate',
+                           units='km')
+        y_coord = DimCoord(np.arange(3), 'projection_y_coordinate',
+                           units='km')
+        self.plugin.topography = iris.cube.Cube(
+            data, long_name="topography", units="m",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+
+    def test_basic(self):
+        """Test outputs are cubes"""
+        gradx, grady = self.plugin._orography_gradients()
+        self.assertIsInstance(gradx, iris.cube.Cube)
+        self.assertIsInstance(grady, iris.cube.Cube)
+
+    def test_values(self):
+        """Test output values and units"""
+        expected_gradx = np.array([[0.12333333, 0.33, 0.53666667],
+                                   [0.2, 0.33333333, 0.46666667],
+                                   [0.27666667, 0.33666667, 0.39666667]])
+
+        expected_grady = np.array([[0.15833333, 0.175, 0.19166667],
+                                   [0.035, 0.03833333, 0.04166667],
+                                   [-0.08833333, -0.09833333, -0.10833333]])
+        gradx, grady = self.plugin._orography_gradients()
+        self.assertArrayAlmostEqual(gradx.data, expected_gradx)
+        self.assertArrayAlmostEqual(grady.data, expected_grady)
+        for cube in [gradx, grady]:
+            self.assertEqual(cube.units, '1')
+
+
+class Test__regrid_variable(IrisTest):
+    """Test the _regrid_variable method"""
+
```

Comment:
Please can you add a test_null function to ensure that a cube that a cube that should not be changed is left unchanged?

Reply:
OK